model_processing/crates/webify_models/src/image_processing/convert_to_png.rs

```diff
@@ -0,0 +1,135 @@
+//! Converts an image file to PNG, or skips if it's already PNG
+
+use std::{fs, io::Error, panic, result::Result};
+
+use image::io::Reader as ImageReader;
+use image::ImageFormat::Tiff;
+
+use crate::image_processing::Image;
+
+/// Convert the specified image to a PNG version
+pub fn convert_to_png(mut image: Image) -> Result<Image, Error> {
+    if image.extension == "tif" {
+        return Ok(image); // Skip tif!
+    }
+    let image_reader = match ImageReader::open(image.path.clone()) {
```

Comment:
You should be able to avoid the `clone()` here by using a borrow. In fact, you could refactor this method a little bit by destructuring borrows from the `Image` struct like this:

![image](https://user-images.githubusercontent.com/5523172/108641728-b3a66200-7455-11eb-92fe-73928c311cf2.png)

I don't know if that appeals to you. That's usually what I do.

Reply:
I hear `clone` is a bad idea in general, so I'll give it a shot!